### DISCUSSION TITLE
Upgrading Rook-Ceph on EKS Cluster with managed node groups

### DISCUSSION THREAD

**Discussion by Ajay250496** [rook/rook#10209] [General]
We have been testing Rook on an EKS cluster in Dev and it has been a great tool that provides all the features we need with regards to data storage, but in the event that we need to update the cluster itself to a new version (e.g 1.20 to 1.21) what would be the process?

Generally we leverage [EKS Managed Node groups](https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html) which automatically cordons, and drains each nodes while pods get deployed onto the new available nodes. Would this be safe for rook? Are there other considerations when upgrading the cluster and cycling in new nodes?

**Comment by travisn** [comment]
@Ajay250496 The critical point for replacing the nodes is that the OSDs and mons are portable, which is done by following the cluster-on-pvc.yaml example. Is that how you deployed? 

Rook can handle the node drains with PDBs, which is explained in [this design doc](https://github.com/rook/rook/blob/master/design/ceph/ceph-managed-disruptionbudgets.md).

**Reply by Ajay250496** [reply]
I am indeed using the cluster-on-pvc.yaml.

This is a great read. 

I need to delve deeper into the PDB to figure out how I can get it to work with the managed node groups. I created a small rook cluster, didnt even have data on it and tried to do an managed node group upgrade but found that that I ran into a "PodEvictionFailure". Unfortunately I don't really have a lot of details as EKS performs this action for you so I'll have to look into the logging there but i did notice that the rook OSD pods did NOT go into a pending state which would mean the pod disruption budget didn't get set (forgive me if this is completely wrong I havent read up on this yet). I may have to resort to manually draining nodes individually to get the OSDs to move over to new nodes gracefully...

**Reply by travisn** [reply]
The rook pdbs will essentially allow the drain of an entire zone (or other [topology](https://rook.github.io/docs/rook/v1.9/ceph-cluster-crd.html#osd-topology)), assuming you have zone labels on the nodes. Does a managed node group correspond to a zone or other topology?
